code/starter_script - Copy.R

```diff
@@ -1,62 +1,46 @@
 # an example of paired predictions and observations
 library(dplyr)
 dat <- readRDS('data/predicted_observed_temps.rds')
-head(dat)
-
 # note that SNTemp predictions (uncalibrated process model)
 # have predictions for all places/times
 # here you just have the test period from the hybrid model (rgcn2_full_temp_c)
 # and sparse observations (some reaches measured, some not)
-
-summary(dat)
-str(dat)
-# example RMSE calculation
-# https://github.com/USGS-R/delaware-model-prep/blob/master/3_predictions/src/munge_predictions.R#L103-L113
-
 # removing rows with na values. 
+
 rgnc_dat_filter <- dat %>% filter(!is.na(temp_c), !is.na(rgcn2_full_temp_c))
 str(rgnc_dat_filter)
 summary(rgnc_dat_filter)
 rgnc_by_seg <- rgnc_dat_filter %>% group_by(seg_id_nat)
+
 ## Absolute Residual Error: calculating Mean Absolute Error (MAE) metric to compare both models prediction data vs observed data. We found the mae by finding the sum of the absolute value difference in predicted and observed temperature. Then divided the sum by the n row in grouped data.
 
 cal_mae <- function(observe_data, predict_data) {
-    mae = (mean(abs(observe_data - predict_data), na.rm = TRUE ))
+    mae = round((mean(abs(observe_data - predict_data), na.rm = TRUE )), digits = 2)
     return(mae)
 }
-
-mae_process_model <- rgnc_by_seg %>% 
-  summarize(MAE_Process_Model = round(cal_mae(temp_c, sntemp_temp_c), 2))
-
-mae_hybrid_model <- rgnc_by_seg %>%
-  summarize(MAE_Hybrid_Model = round(cal_mae(temp_c, rgcn2_full_temp_c), 2))
+mae <- rgnc_by_seg %>%  
+  summarize(MAE_Process_Model = cal_mae(temp_c, sntemp_temp_c),
+            MAE_Hybrid_Model = cal_mae(temp_c, rgcn2_full_temp_c))
 
 # Absolute Residual Error: Root Mean Square Error (RMSE) square the difference in predicted and observed temperature.  sum the squared values, then divided by n of row. Finally,took the square root
 cal_rmse <- function(observe_data, predict_data){
```

Comment:
I think you should include n_digits as an argument to the function. The default could be set to 2 so you don't have to use it often. The function line would be:

`cal_rmse <- function(observe_data, predict_data, n_digits = 2) {`

and in the function would look like:

`rmse <- round(sqrt(mean((observe_data - predict_data) ^2, na.rm = TRUE )), digits = n_digits)`